cpu/native/native_cpu.c

```diff
@@ -187,30 +213,40 @@ void isr_thread_yield(void)
 {
     DEBUG("isr_thread_yield\n");
 
+    if (_native_sigpend > 0) {
+        DEBUG("isr_thread_yield(): handling signals\n\n");
+        native_irq_handler();
```

Comment:
This may lead to running the wrong thread. After `isr_cpu_switch_context_exit()` is called, the previous thread is restored before `sched_run()` is getting called.